### ISSUE TITLE
Add constrained mixture material model

### ISSUE THREAD

**Issue by mrp089** [SimVascular/svMultiPhysics#2]
@schwarz-e has a [`C++` implementation](https://github.com/StanfordCBCL/svFSG/tree/master/FolderExecutables) of the constrained mixture model (CMM) that takes the deformation gradient and returns the stress tensor and tangential stiffness matrix. Currently, this is called from a Python code.

We have a unique opportunity to take this CMM implementation and put it into `svFSIplus` as a material model, which hasn't been done before in any large-scale FEM code. The challenge is that we need to store a bunch of information on each integration point, which is currently all implemented in a `vessel` struct. Also, we need to test if the linearization of the stress tensor is correct.

I think this could be an interesting first `svFSIplus` project for me.

PS: Happy issue #2!

**Comment by ktbolt** [comment]
No recent activity, closing for now.

**Comment by alisonmarsden** [comment]
I think @Marisa Schmidt ***@***.***> is working on this still!


From: Dave Parker ***@***.***>
Date: Wednesday, February 5, 2025 at 10:43 AM
To: SimVascular/svMultiPhysics ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [SimVascular/svMultiPhysics] Add constrained mixture material model (Issue #2)

No recent activity, closing for now.

**Comment by alisonmarsden** [comment]
Our postdoc working on FSG models.    (I don’t know her GitHub handle).


From: Dave Parker ***@***.***>
Date: Wednesday, February 5, 2025 at 10:49 AM
To: SimVascular/svMultiPhysics ***@***.***>
Cc: Alison Lesley Marsden ***@***.***>, Mention ***@***.***>
Subject: Re: [SimVascular/svMultiPhysics] Add constrained mixture material model (Issue #2)

@alisonmarsden<https://github.com/alisonmarsden> Who is marisa ?

**Comment by ktbolt** [comment]
@msbazzi Are you working on this? This Issue has not been updated for over a year so if you are still working on this then update with Issue with recent progress or any discussions you have had.

**Comment by msbazzi** [comment]
@ktbolt yes, I have been working on this! I am sorry that I haven't updated this issue. Most of the progress I have made so far is in transferring @schwarz-e implementation of spatially distributed material parameters (issue #77) from Fortran to svMultiphysics, and improving and testing the [current framework ](https://github.com/StanfordCBCL/svFSG/tree/master/FolderExecutables).

Bringing this framework to svMultiPhysics is the next step.

**Comment by ktbolt** [comment]
@msbazzi It is important to communicate to other team members what you are doing so they can provide feedback. Issue updates also provide a history of code development decisions that may be useful later. I typically describe my implementation strategy and what code changes/additions I am making.

**Comment by msbazzi** [comment]
@ktbolt Noted!! I will do a better job updating the issues, and them using as a channel for communication with the team!
